proposals/nnnn-atf-selenium-support.md

```diff
@@ -0,0 +1,236 @@
+
+#### Extend SendResponse
+
+After handling the request, the HMI will automatically respond with a success response. ATF will need to catch this response from the HMI and match it with the parameters included in the test. If the parameters match, ATF will forward the message to SDL core and the test will continue. To implement this behavior, the SendResponse() method will need to be extended for selenium compatible test scripts.
+
+#### Extend SendNotification
+
+Most notifications sent by the HMI are triggered as a result of a user interaction a with the HMI. Examples of these events are pressing a menu command, or consenting to app permissions.
+
+```lua
+local function onCommand()
+  common.getHMIConnection():SendNotification("UI.OnCommand", { cmdID = 20, appID = common.getHMIAppId() })
+  common.getMobileSession():ExpectNotification("OnCommand", { cmdID = 20, triggerSource= "MENU" })
+end
+```
+
+SendNotification should be extended to notify the selenium web driver it should simulate a user action if applicable. In this case it would be selecting menu command with cmdID = 20. This will trigger the HMI to send the `UI.OnCommand` RPC. ATF will need to listen for this notification and match it with the parameters listed in the test. ATF will then forward the notification from the HMI to SDL Core after validating the msg contents.
+
+#### Extend SendRequest
+
+Similar to SendNotificaiton, ATF sometimes uses SendRequest to simulate an event that was triggered by a user action.
+
+```lua
+function m.app.activate(pAppId)
+  if not pAppId then pAppId = 1 end
+  local requestId = m.hmi.getConnection():SendRequest("SDL.ActivateApp", { appID = m.app.getHMIId(pAppId) })
+  m.hmi.getConnection():ExpectResponse(requestId)
+  m.mobile.getSession(pAppId):ExpectNotification("OnHMIStatus", { hmiLevel = "FULL", systemContext = "MAIN" })
+  if m.mobile.getAppsCount() > 1 then m.run.wait(m.minTimeout) end
+end
+```
+
+In this example the HMI is requesting to activate an application via `SDL.ActivateApp`. This action implies the user has selected the app from the app list, therefore the selenium application needs to simulate a user action. SendRequest would be extended to notify the selenium web driver to click an app on the app list to activate the app.
+
+#### Extend ExpectResponse
+
+Using the previous activate app example, the ExpectResponse method should be extended to be able to validate the state of the HMI changed post activating an app.
+
+### RPC Switching
+
+Because there is a mix of behavior for overwriting the previously mentioned methods in terms of requiring a simulated user action vs verifying the content displayed on the HMI, ATF should implement a structure that defines what actions the selenium web driver must take in the extended atf methods.
```

Comment:
```suggestion
Because there is a mix of behavior for overwriting the previously mentioned methods in terms of requiring a simulated user action vs verifying the content displayed on the HMI, ATF should implement a structure that defines which actions the Selenium WebDriver must take in the extended ATF methods.
```